src/plugins/data_views/common/data_views/data_view.ts

```diff
@@ -68,188 +51,32 @@ export interface TimeBasedDataView extends DataView {
 /**
  * Data view class. Central kibana abstraction around multiple indices.
  */
-export class DataView implements DataViewBase {
-  /**
-   * Saved object id
-   */
-  public id?: string;
-  /**
-   * Title of data view
-   * @deprecated use getIndexPattern instead
-   */
-  public title: string = '';
-  /**
-   * Map of field formats by field name
-   */
-  public fieldFormatMap: FieldFormatMap;
-  /**
-   * Only used by rollup indices, used by rollup specific endpoint to load field list.
-   */
-  public typeMeta?: TypeMeta;
+export class DataView extends AbstractDataView implements DataViewBase {
```

Comment:
It's unfortunate we can't implement `DataViewBase` in `AbstractDataView` instead, but I'm guessing that's because of the `fields` array? It makes the data view abstraction a little harder to understand IMO, but I'm not sure how to get around it.

Reply:
Yeah, DataViewBase needs to have a field list. I could investigate renaming it, it might not be used in too many places.

Reply:
I think we should be fine with this to start and we can continuing evolving it in subsequent PRs.